src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.cs

```diff
@@ -851,5 +867,22 @@ private enum CurrentOperationType
             Write,
             Flush
         }
+
+        private WindowsPrincipal GetWindowsPrincipal()
+        {
+            var hr = NativeMethods.http_get_authentication_primary_token(_pHttpContext, out IntPtr token);
+            if (hr == 0 && token != IntPtr.Zero)
+            {
+                var identity = new WindowsIdentity(token);
```

Comment:
We may need to set the auth type parameter.
https://github.com/aspnet/IISIntegration/issues/231
https://github.com/aspnet/HttpSysServer/blob/7f96eab775d721a3bf8d05be446086dab4b45aae/shared/Microsoft.AspNetCore.HttpSys.Sources/RequestProcessing/NativeRequestContext.cs#L203-L232